code/modules/food_and_drinks/machinery/dinner_button.dm

```diff
@@ -0,0 +1,25 @@
+/obj/machinery/foodbutton
+	name = "dinner button"
+	desc = "Casts an announcement to the station, reminding the crew about food."
+	icon = 'icons/obj/service/kitchen.dmi'
+	icon_state = "button_supper"
+	layer = OBJ_LAYER
+	anchored = TRUE
+	pass_flags = PASSTABLE // Able to place on tables
+	var/button_used = FALSE // only one announcement per shift
+	var/button_sound = 'sound/machines/pda_button2.ogg'
+
+/obj/machinery/foodbutton/Initialize(mapload)
+	. = ..()
+
+/obj/machinery/foodbutton/attack_hand(mob/living/user)
+	. = ..()
+	announce()
+
+/obj/machinery/foodbutton/proc/announce(mob/living/user)
+	if(button_used)
```

Comment:
```suggestion
	if(!allowed(user))
		balloon_alert(user, "access denied!")
		return
	if(button_used)
		balloon_alert(user, "already used!")
```
It should be locked to those with proper kitchen access to prevent random asshats from untimely using it.